### PR TITLE
Replace RuleAlteredJobContext dependency to JobConfiguration in DataConsistencyCheckerImpl

### PR DESCRIPTION
**Pull request by sandynz**
Purpose:
- Simplify `DataConsistencyCheckerImpl` dependency (`JobConfiguration` is part of `RuleAlteredJobContext`), cleaner code

Changes proposed in this pull request:
- Replace `RuleAlteredJobContext` dependency to `JobConfiguration` in `DataConsistencyCheckerImpl`

### CODE REVIEW

**Comment by terrymanu** [comment]
The license check is failed, please fix it

**Comment by sandynz** [comment]
> The license check is failed, please fix it

`WorkerIdNode` license was added in master branch. Master branch merged now.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/main/java/org/apache/shardingsphere/data/pipeline/api/config/rulealtered/HandleConfiguration.java [hunk 1]
```diff
@@ -51,6 +51,9 @@
     
     private List<String> jobShardingDataNodes;
     
+    /**
+     * Logic tables. Separated by ','. e.g. t_order,t_order_item.
+     */
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/main/java/org/apache/shardingsphere/data/pipeline/api/config/rulealtered/HandleConfiguration.java:56]: There is unnecessary to add JavaDoc for private filed

Comment (sandynz) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/main/java/org/apache/shardingsphere/data/pipeline/api/config/rulealtered/HandleConfiguration.java]: JavaDoc removed.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**